src/draw_partial.jl

```diff
@@ -0,0 +1,474 @@
+        if dp_state.cur_len_partial >= dp_state.target_len_partial
+            return
+        end
+        if dp_state.target_len_partial - dp_state.cur_len_partial < dist
+            d = (dp_state.target_len_partial - dp_state.cur_len_partial) / dist
+            circle(corner1, corner1 + d * vec, :fill)
+            dp_state.cur_len_partial = dp_state.target_len_partial
+            dp_state.draw_state = false
+        else
+            circle(corner1, corner1 + dist * vec, :fill)
+            dp_state.cur_len_partial += dist
+        end
+        clipreset()
+        newpath()
+    end
+end
+
+
+function overdub(c::ctx_partial, ::typeof(Luxor.fillpreserve), args...)
+    global dp_state
+    current_path = storepath()
+    if dp_state.draw_state == false
+        return
+    else
+        poly1, _ = pathtopoly(:yes)
+        if length(vcat(poly1...)) > 2
+            bbox = BoundingBox(vcat(poly1...))
+            dist = distance(bbox[1], bbox[2])
+        else
+            bbox = BoundingBox([O, O])
+            dist = 0
+        end
+        corner1 = bbox[1]
+        corner2 = bbox[2]
+        vec = corner2 - corner1
+        do_action(:clip)
+        if dp_state.cur_len_partial >= dp_state.target_len_partial
+            return
+        end
+        if dp_state.target_len_partial - dp_state.cur_len_partial < dist
+            d = (dp_state.target_len_partial - dp_state.cur_len_partial) / dist
+            circle(corner1, corner1 + d * vec, :fill)
+            dp_state.cur_len_partial = dp_state.target_len_partial
+            dp_state.draw_state = false
+        else
+            circle(corner1, corner1 + dist * vec, :fill)
+            dp_state.cur_len_partial += dist
+        end
+        clipreset()
+    end
+    drawpath(current_path)
+end
+
+#strokepath
+function overdub(c::ctx_partial, ::typeof(Luxor.strokepath), args...)
+    global dp_state
+    dp_state.stroke_count += 1
+
+    polys, co_states = pathtopoly(:yes)
+    if dp_state.draw_state == false
+        newpath()
+        return
+    end
+    newpath()
+    pdists = CURRENT_OBJECT[1].opts[:polydistances][dp_state.stroke_count]
+    for (poly_i, co, pdist_i) in zip(polys, co_states, pdists)
+        if dp_state.cur_len_partial >= dp_state.target_len_partial
+            return
+        else
+            #since its a stroke we dont need path after this
+            nextpolylength = polyperimeter(poly_i, closed = co)
+            if dp_state.cur_len_partial + nextpolylength < dp_state.target_len_partial
+                poly(poly_i, close = co)
+                dp_state.cur_len_partial += nextpolylength
+            else
+                if length(poly_i) >= 2
+                    frac =
+                        (dp_state.target_len_partial - dp_state.cur_len_partial) /
+                        nextpolylength
+                    poly(polyportion(poly_i, frac, closed = co, pdist = pdist_i))
+                    #poly(polyportion(poly_i, frac, closed = co))
```

Comment:
there seems to be no difference in performance if i switch this line for the line above (with `pdist = pdist_i`) ;
atleast for the example i tried (hilbert curve)
we would expect passing the polydistances to polyportion to make it faster .